makelib/helm.mk

```diff
@@ -84,6 +88,20 @@ $(HELM_OUTPUT_DIR)/$(1)-$(HELM_CHART_VERSION).tgz: $(HELM_HOME) $(HELM_OUTPUT_DI
 	fi
 	@$(OK) helm package $(1) $(HELM_CHART_VERSION)
 
+helm.generate.$(1): $(HELM_HOME) $(HELM_DOCS)
+	@if [ "$(HELM_DOCS_ENABLED)" != "true" ]; then \
+		$(OK) helm-docs $(1) [disabled]; \
+	else \
+		$(INFO) helm-docs $(1); \
+		$(MAKE) VERSION="master" helm.prepare.$(1); \
+		cd $(HELM_CHARTS_DIR)/$(1) && $(HELM_DOCS); \
+		rm -f $(HELM_CHARTS_DIR)/$(1)/values.yaml; \
```

Comment:
you mean only the `rm -f` part, right?

Reply:
No, helm docs invocation. If we move it there, we wouldn't already need the 1st and 3rd lines highlighted above.

Reply:
gave it a shot here: https://github.com/upbound/build/pull/230/commits/221966359795c8dd5b968bdbfcf1fa4e9e6bfa76

Reply:
I'm definitely not a Makefile expert, but I wanted this to be run on `make reviewable` and `make check-diff`, the former runs the prepare target only as a requirement for the lint, while the latter does not currently run the prepare target. While they both run directly the generate target.

Reply:
ok, with the latest changes it will continue working as it is currently if nothing is declared by users, but if the needed values are set as [in the Makefile here](https://github.com/crossplane/crossplane/commit/1b4aeaf7e5e91fa07be709d93ef14d1bff0d4ba3), it will skip the templating and generate the docs as needed, the commit also directly address https://github.com/crossplane/crossplane/issues/3994.